src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANGE = 100;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static int index;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private String figureType;
+    private Figure figure;
+    private String figureColor;
+
+    public Figure getRandomFigure() {
+        index = random.nextInt(FigureType.values().length);
+        figureType = FigureType.values()[index].name();
+        figureColor = colorSupplier.getRandomColor();
+        switch (figureType) {
+            case "SQUARE":
+                figure = new Square(random.nextInt(RANGE));
+                figure.setColor(figureColor);
+                return figure;
```

Comment:
```suggestion
                figure = new Square(random.nextInt(RANGE));
                figure.setColor(figureColor);
                return figure;
```

move creating each figures to the separate method, like getRandomSquare()

Reply:
I can't do this. Because in each CASE block I create a different figure! So, in this case I should made as many separate methods as CASE blocks I have. I don't think that it's good idea. I can put all CASE block in separate method...but I really don't see the point in this too...